dbos/dbos.go

```diff
@@ -119,87 +215,90 @@ func Initialize(inputConfig Config) error {
 	// Create the system database
 	systemDB, err := NewSystemDatabase(config.DatabaseURL)
 	if err != nil {
-		return newInitializationError(fmt.Sprintf("failed to create system database: %v", err))
+		return nil, newInitializationError(fmt.Sprintf("failed to create system database: %v", err))
 	}
 	initExecutor.systemDB = systemDB
 	logger.Info("System database initialized")
 
-	// Set the global dbos instance
-	dbos = initExecutor
-
-	return nil
+	return initExecutor, nil
 }
 
-func Launch() error {
-	if dbos == nil {
-		return newInitializationError("DBOS instance not initialized, call Initialize first")
+func (c *dbosContext) Launch() error {
+	if c.launched {
+		return newInitializationError("DBOS is already launched")
 	}
+
 	// Start the system database
-	dbos.systemDB.Launch(context.Background())
+	c.systemDB.Launch(context.Background())
 
 	// Start the admin server if configured
-	if dbos.config.AdminServer {
-		adminServer := newAdminServer(_DEFAULT_ADMIN_SERVER_PORT)
+	if c.config.AdminServer {
+		adminServer := newAdminServer(c, _DEFAULT_ADMIN_SERVER_PORT)
 		err := adminServer.Start()
 		if err != nil {
 			logger.Error("Failed to start admin server", "error", err)
 			return newInitializationError(fmt.Sprintf("failed to start admin server: %v", err))
 		}
 		logger.Info("Admin server started", "port", _DEFAULT_ADMIN_SERVER_PORT)
-		dbos.adminServer = adminServer
+		c.adminServer = adminServer
 	}
 
 	// Create context with cancel function for queue runner
-	ctx, cancel := context.WithCancel(context.Background())
-	dbos.queueRunnerCtx = ctx
-	dbos.queueRunnerCancelFunc = cancel
-	dbos.queueRunnerDone = make(chan struct{})
+	// FIXME: cancellation now has to go through the DBOSContext
+	ctx, cancel := context.WithCancel(c.ctx)
+	c.queueRunnerCtx = ctx
+	c.queueRunnerCancelFunc = cancel
+	c.queueRunnerDone = make(chan struct{})
```

Comment:
In a next PR I'll tighten up the usage of DBOSContext to manage its owned resources